src/utils/maak_protocol.R

```diff
@@ -0,0 +1,263 @@
+  assert_that(!(korte_titel %in% korte_titels), 
+              msg = "De opgegeven korte titel bestaat al. Geef een korte titel die nog niet bestaat.")
+  
+  protocol_code <- paste0(protocol_type, "-", protocol_nummer)
+  folder_name <- paste0(protocol_code, "_", korte_titel)
+  folder_name <- tolower(folder_name)
+  
+  # directory setup
+  if (missing(projectnaam)) {
+    path_to_protocol <- file.path(project_root, "src", "thematic", thematiek, 
+                                  folder_name)
+  } else {
+    path_to_protocol <- file.path(project_root, "src", "project", projectnaam, 
+                                  thematiek, folder_name)
+  }
+  
+  # check for existence of the folder
+  if (dir.exists(path_to_protocol)) {
+    stop(sprintf(paste0("De protocol repository bevat reeds ",
+                        "een map voor %s!"), path_to_protocol))
+  } else {
+    # create a new directory
+    dir.create(file.path(path_to_protocol), 
+               recursive = TRUE)
+  }
+  
+  # move all files from the template folder and add info
+  files <- list.files(path = path_to_template, 
+                      full.names = FALSE,
+                      recursive = FALSE, 
+                      pattern = protocol_type)
+  
+  for (filename in files) {
+    print(paste0("Preparing: ", filename))
+    new_file_path <- file.path(path_to_protocol, filename)
+    original_file_path <- file.path(path_to_template, filename)
+   
+    # create new file
+    file.create(new_file_path)
+    file.rename(from = new_file_path, 
+                to = file.path(path_to_protocol, paste0(folder_name, ".Rmd")))
+    new_file_path <- file.path(path_to_protocol, paste0(folder_name, ".Rmd"))
+    
+    # add original file content from reference, changing values
+    original_file <- file(original_file_path, "r")
+    original_file_content <- readLines(original_file)
+    data <- list(titel = titel,
+                 subtitel = subtitel,
+                 auteurs = auteurs,
+                 datum = datum,
+                 reviewers = reviewers,
+                 documentbeheerder = documentbeheerder,
+                 revisie = revisie,
+                 procedure = protocol_code
+                 )
+    writeLines(purrr::map_chr(original_file_content,
+                              whisker::whisker.render,
+                              data),
+               new_file_path)
+    close(original_file)
+  }
+}
+
+
+
+
+#' @title Hulpfunctie voor het oplijsten van bestaande protocolnummers
+#'
+#' @param protocol_type
+#'
+#' @return Een vector met in gebruik zijnde protocolnummers (als character) voor een bepaald protocoltype.
+#'
+#' @importFrom rprojroot find_root
+#' @importFrom rprojroot is_git_root
+#' @importFrom assertthat
+#' assert_that
+#' is.string
+#' @importFrom stringr
+#' str_detect
+#' str_extract
+#' @examples
+#' geef_protocolnummers()
+#'
+geef_protocolnummers <- function(protocol_type = c("svp", "sip", "sap", "sop")) {
+  require(rprojroot)
```

Comment:
```suggestion
  stopifnot(require(rprojroot))
```

stops the function when the package is not available